### PR TITLE
Add more detailed notes about Inject and Join

### PR DESCRIPTION
**Pull request by bensigelman**
Most of the writing is new, though some was moved from `spec.md`.

### CODE REVIEW

File: propagation.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+For `Inject` and `Join` to be useful, all of the following must be true:
+
+- Per the above, the [OpenTracing user](/use-cases/#stepping-back-who-is-opentracing-for) handling cross-process trace propagation must not need to write OpenTracing-implementation-specific code
+- OpenTracing implementations must not need special handlers for every known inter-process communication mechanism: that's far too much work, and it's not even well-defined
+- That said, the propagation mechanism should be extensible for optimizations
+
+## The basic approach
+
+Any Span in a trace may be **Injected** into a **Carrier**. Similarly, given a **Carrier**, an injected trace may be **Joined**, yielding a new Span.
```
Comment (yurishkuro) [inline propagation.md]: I think we should first define the "carrier" being an abstraction of the IPC data exchange mechanism

**Comment by yurishkuro** [comment]
one comment, otherwise lgtm

File: propagation.md (hunk 1 above)
Comment (bensigelman) [inline propagation.md]: Good suggestion... I went a bit further, hopefully this will clarify things. It's an incremental improvement if nothing else.

Comment (bensigelman) [inline propagation.md]: (https://github.com/opentracing/opentracing.github.io/pull/84/commits/f73e49ecf5c12293f056d7c3ac36d85a3d6106f9)

**Comment by yurishkuro** [comment]
Fixes #79

**Pull request merged**